src/d3d12/d3d12_command_queue.cpp

```diff
@@ -33,9 +33,16 @@ namespace wr::d3d12
 
 		cmd_queue->m_native->ExecuteCommandLists(native_lists.size(), native_lists.data());
 
-		fence->m_fence_value++;
-		Signal(fence, cmd_queue);
+		if (fence != nullptr)
+		{
+			fence->m_fence_value++;
+			Signal(fence, cmd_queue);
+		}
+	}
 
+	void WaitFor(CommandQueue * cmd_queue, Fence * fence)
+	{
+		cmd_queue->m_native->Wait(fence->m_native, fence->m_fence_value);
```

Comment:
Why did you need this? `Fence::WaitFor` should do the job right? And it won't require knowledge about the queue.

Reply:
This makes the queue wait on the fence, instead of the program. It allows you to synchronize multiple queues without having to halt the cpu.